build-system/compile/single-pass.js

```diff
@@ -683,7 +703,13 @@ function compile(flagsArray) {
       .on('error', (err) => handleSinglePassCompilerError(err))
       .pipe(gulpIf(!argv.pseudo_names, checkForUnknownDeps()))
       .on('error', reject)
-      .pipe(sourcemaps.write('.'))
+      .pipe(
+        sourcemaps.write('.', {
+          sourceRoot: isProdBuild
+            ? `https://raw.githubusercontent.com/ampproject/amphtml/${internalRuntimeVersion}/`
+            : 'http://localhost:8000/',
```

Comment:
both `localhost` and `port` are configurable locally, should this account for the configurability?

Reply:
I like @kristoferbaxter's suggestion. If the URL generation is broken out into its own method, I would also be able to use it for https://github.com/ampproject/amphtml/pull/27661 .

Reply:
Agree with moving this to a separate function. In addition, see my above comment about changing the logic and removing `isProdBuild`.